ingestors/support/ocr.py

```diff
@@ -45,6 +48,81 @@ def extract_ocr_text(self, data, languages=None):
         return stringify(text)
 
 
+class ZBarDetectorService(object):
+    THRESHOLDS = list(range(32, 230, 32))
+
+    def _enhance_image(self, image, threshold=127):
+        width, height = image.size
+
+        # Convert to grayscale using Pillow
+        image = image.convert("L")
+
+        # Apply Gaussian blur to reduce noise
+        image = image.filter(ImageFilter.GaussianBlur(3))
+
+        # Apply threshold
+        image = image.point(lambda p: 255 if p > threshold else 0)
+
+        # Dilatate the image
+        image = image.filter(ImageFilter.MaxFilter(3))
+
+        # Erode the image
+        image = image.filter(ImageFilter.MinFilter(3))
+
+        # Resize the image to make the QR code larger
+        new_size = map(lambda x: x * 2, image.size)
+        image = image.resize(new_size, resample=Image.Resampling.BILINEAR)
+
+        # Last round of gaussian blur
+        image = image.filter(ImageFilter.GaussianBlur(5))
+        return image
+
+    def _serialize_zbar_result(self, result):
+        return "\n".join(
+            [
+                "",
+                "--- {} CODE ---".format(result.type),
+                "QUALITY: {}".format(result.quality),
+                "ORIENTATION: {}".format(result.orientation),
+                "POSITION: {}".format(list(result.rect)),
+                "DATA: {}".format(result.data.decode("utf-8")),
+            ]
+        )
+
+    def _results_to_text(self, results):
+        return "---\n".join([self._serialize_zbar_result(result) for result in results])
+
+    def _try_best(self, image):
+        results = pyzbar.decode(image)
+        # Found it at first try
+        if len(results) > 0:
+            log.info("OCR: zbar found (%d) results at first shot", len(results))
+            return results
+
+        log.info("OCR: zbar ehnahcing image")
+        # Try with our enhance logic
+        for threshold in self.THRESHOLDS:
+            log.info("OCR: zbar applying threshold %d", threshold)
+            # Headsup: preserve the original image
+            new_image = self._enhance_image(image, threshold=threshold)
+            results = pyzbar.decode(new_image)
+
+            if len(results) > 0:
+                log.info(
+                    "OCR: zbar found (%d) results with threshold=%d",
+                    len(results),
+                    threshold,
+                )
+                return results
+
+        # no results found then
+        return []
+
+    def extract_barcodes(self, image):
```

Comment:
- [ ] I'd appreciate adding type hints here. Assuming `image` is a `PIL.Image` and the return type is a `str`?
- [ ] (minor) This being public it would be great to add a docstring saying what it does, since at first glance I wouldn't necessarily expect this to return text. (Perhaps `extract_text_from_barcodes` is more appropiate?